coverage.py

```diff
@@ -1104,6 +1111,22 @@ def set_presentation_ready(self, identifier):
         return identifier
 
 
+class CollectionCoverageProviderJob(Job):
+
+    def __init__(self, collection, provider_class, item_offset,
+        **provider_kwargs
+    ):
+        self.collection = collection
```

Comment:
I believe there's a way to create the provider at the thread/`DatabaseWorker` level, which may greatly simplify the work that each `Job` does. Initial tests lead to integrity errors, so I'm leaving it for a future PR.

Reply:
I've attempted this twice now, and both resulted in lots of data integrity errors (usually around the 'id' set at db insert time for various tables).